### PR TITLE
Exclusive Fields Validator Plugin

### LINKED ISSUE #257: Create the Exclusive Fields validation plugin
Create the "Exclusive Fields" validation plugin. This validation plugin verifies that the definition being validated only defines one of the exclusive fields provided via arguments

## AC:
- [ ] The validator plugin implements the following plugin hooks
  - [ ] `get_plugin_aac_definitions()`
  - [ ] `register_validators()`
- [ ] The validator contains acceptance scenarios for
  - [ ]  validation success
  - [ ]  validation failure
- [ ] The validation is applied to the `ext:` definition 
  - [ ] `ext:` requires that  "enumExt" and "dataExt" are the exclusive fields
- [ ] Validation fails if an `ext:` definition contains both enumExt and dataExt fields
- [ ] Validation is applied to the `extension` AaC definition

### PR DESCRIPTION
**Pull request by Coffee2Bits**
Closes #257 

* Add the exclusive fields validator plugin
* Added exclusive fields to the extension definition

### CODE REVIEW

File: python/src/aac/plugins/validators/exclusive_fields/_exclusive_fields.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+def validate_exclusive_fields(definition_under_test: Definition, target_schema_definition: Definition, active_context: LanguageContext, *validation_args) -> ValidatorResult:
+    """
+    Validates that the none of the fields are simultaneously defined.
+
+    Args:
+        definition_under_test (Definition): The definition that's being validated.
+        target_schema_definition (Definition): A definition with applicable validation.
+        active_context (LanguageContext): The active context.
+        *validation_args (string): The list of exclusive fields.
```
Comment (cvchaparro) [inline python/src/aac/plugins/validators/exclusive_fields/_exclusive_fields.py:18]:
```suggestion
        *validation_args (list[string]): The list of exclusive field names.
```

**Review by cvchaparro — CHANGES_REQUESTED** [review]
Couple minor changes - nice work, otherwise.

File: python/src/aac/plugins/validators/exclusive_fields/_exclusive_fields.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+
```
[the remark this answers is not in the document]
Comment (Coffee2Bits) [reply python/src/aac/plugins/validators/exclusive_fields/_exclusive_fields.py]: Addressed

File: python/src/aac/plugins/validators/exclusive_fields/_exclusive_fields.py (hunk 1 above)
Comment (Coffee2Bits) [reply python/src/aac/plugins/validators/exclusive_fields/_exclusive_fields.py]: Addressed

**Review by cvchaparro — APPROVED** [review]
Nice work

**Pull request merged**